aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -136,34 +136,69 @@ constexpr auto calc_io_size(){
 template <int vec_size, typename func_t, typename array_t>
 C10_LAUNCH_BOUNDS_1(num_threads())
 __global__ void vectorized_elementwise_kernel(int N, func_t f, array_t data) {
-  using traits = function_traits<func_t>;
-  constexpr auto io_size = calc_io_size<func_t>();
-  int remaining = N - io_block_work_size<io_size>() * blockIdx.x;
-
-  if (remaining < io_block_work_size<io_size>()) { // if this block handles the reminder,
-                                       // just do a naive unrolled loop
-    auto input_calc = TrivialOffsetCalculator<traits::arity>();
-    auto output_calc = TrivialOffsetCalculator<1>();
-    auto loader = memory::LoadWithoutCast();
-    auto storer = memory::StoreWithoutCast();
-    auto policy = memory::policies::unroll<
-        array_t,
-        decltype(input_calc),
-        decltype(output_calc),
-        memory::LoadWithoutCast,
-        memory::StoreWithoutCast,
-        elems_per_thread<io_size>()>(
-        data, remaining, input_calc, output_calc, loader, storer);
-    elementwise_kernel_helper(f, policy);
-  } else { // if this block has a full `block_work_size` data to handle, use
-           // vectorized memory access
+  if constexpr (vec_size == 8) {
+  // To save some build time on CUDA, we are going to utilize vec8 only on SM90+ devices.
+#if defined(USE_ROCM) || ((defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 900))
```

Comment:
Ah, I see. This is where the buildtime is changed. Please allow SM89 devices too if they support them.

Reply:
here, right https://github.com/pytorch/pytorch/blob/0bd2caac55aa518bc1148c5365f42e7946f539a1/aten/src/ATen/native/cuda/CUDALoops.cuh#L220

Reply:
>Just don't instantiate this kernel with vec_size == 8 on older arches, don't add 2 similar codeblocks to the kernel itself

I wish I could do that, the issue is that `__CUDA_ARCH__` is available only inside kernels. The place where @eqy pointed is a host side code.

Reply:
Are the binary size savings really worth it? This adds considerable complexity and maintenance burden to the kernel

Reply:
@ngimel this was raised as it was adding many (dozens?) of minutes to the windows build time, not due to binary size concerns